static/mixins/polling.jsx

```diff
@@ -6,11 +6,67 @@ var PollingMixin = {
       pollingActive: true
     };
   },
+  pushNotification(){
+    let body;
+    let {app, env, number} = this.props.params;
+    const url = `/deploys/${app}/${env}/${number}`;
 
-  componentWillMount() {
+    if(this.state.task !== undefined){
+      body = `${/[^@]*/.exec(this.state.task.user.name)}'s deploy ${this.state.task.status}`;
+    }else{
+      body = `${/[^@]*/.exec(this.state.deploys[0].user.name)}'s deploy ${this.state.deploys[0].status}`;
+    }
+
+    const options = {
+      body: body,
+      icon: "../static/favicon.png"
+    }
+
+    if (!("Notification" in window)) {
+      alert("This browser does not support system notifications");
+    }
+
+  else if (Notification.permission === "granted") {
+    if(this.state.task != undefined){
+      let notification = new Notification(`${this.state.task.name}`, options)
```

Comment:
Why are these using string formatting stuff? You can just do `new Notification(this.state.task.name, options)`